### DISCUSSION TITLE
VisiData Office Hours tomorrow

### DISCUSSION THREAD

**Discussion by anjakefala** [saulpw/visidata#1906] [Q&A]
Saul will be hosting another drop-in VisiData Office Hours this upcoming Friday, May 26th at 10am PDT!

This is a great chance to bring any questions you might have, discuss any issues you've encountered, brainstorm some feature requests, or simply to share your love for VisiData. Meeting each other can also benefit with us realising the human on the other side of the pixel avatar.

All are welcome to this informal gathering of VisiData enthusiasts, whether you're a seasoned user or just getting started.

Here are the details:

🗓️ When: 2023-05-26, 10:00 PDT (Check your timezone [here](https://everytimezone.com/s/258910d7)) 
📍 Where: https://whereby.com/visidata

Hope to see you there!

**Comment by saulpw** [comment]
# VisiData Office Hours summary (May 2023)

## Updates

- Saul is potentially taking on a contract to develop a custom VisiData app for internal company use, which would mean this would be a another summer of exciting VisiData development.
- Saul will be recording a podcast with @jsvine and @eigenfoo next week on crossword puzzle collections.
- Saul introduced his latest project, [AIPL](https://github.com/saulpw/aipl), which is becoming a batch scripting language with a VisiData-style structure. AIPL scripts should be more easily understandable and modifiable than existing VisiData cmdlogs.

## Tool Recommendations

- [Pyright](), a typechecker for Python, was discussed as a tool that @jreagle believes works more effectively than mypy.
- [Espanso](), a text expander tool

## User Experience and Feedback

- Long-time VisiData user and enthusiast @aborruso came by and expressed his admiration for the work we do <3.
- He highlighted how he and his team used VisiData and his Italian documentation to manage pandemic-related data during the pandemic.
- He has been using DuckDB for more in-depth analysis, and spoke about his work with spatial data and the lack of VisiData-like tools for GIS.

## Potential Enhancements

- The aforementioned contract might need hierarchical display of data, and pop-up dialog forms for input.
- A [VisiData command palette]() is planned for version 3.0, to help users remember commands and make VisiData more easy to start using and to return to after some time away.
- @aborruso expressed the desire for a better Parquet loader and saver in VisiData, citing Parquet as a likely future of open data formats. While a PyArrow loader currently exists, it has not received significant attention yet.
- Can pandas or other Python code be exported from VisiData to reproduce workflows?  Already vdsql can almost do this with Ibis.  Polars is another interesting alternative.
- VisiData can't use Textual for interface components because the architectures are too dissimilar (VisiData actively draws to windows and manages its own interface loop; Textual needs apps to declare interface instead of draw, so it can manage the interface loop).  Also Textual does not support some basic TUI functionality, like Ctrl+Z to suspend, or launching a terminal editor like vim to edit a cell.

**Comment by anjakefala** [comment]
@frosencrantz this time we took notes! =)

**Reply by frosencrantz** [reply]
@anjakefala and @saulpw  That's wonderful, thanks for sharing! 

The links for software mentioned above point back to this thread.  Here are the links:
- https://github.com/microsoft/pyright
- https://github.com/espanso/espanso

I am interested in hearing about how this potential contract might change VisiData. Both hierarchical display of data, and pop-up dialog forms for input would be exciting updates.



I :heart: to hear about @aborruso's experience, and feedback.  Looking forward to hearing the upcoming podcast.

**Reply by aborruso** [reply]
Dear @frosencrantz 

> I ❤️ to hear about @aborruso's experience, and feedback. Looking forward to hearing the upcoming podcast.

I connected late. I had a busy day at work, and I was short on time.
But it was too good for me to be able to tell Saul and Anja how grateful I was to them. And so, I tried to connect, they were still connected, and I could virtually hug them.
Unfortunately, I have bad spoken English and so I mostly listened, and I couldn't say what I wanted properly.

This idea of these meetings is nice, and I think it will do some good for VisiData and all its users.

The only sad thing? That you weren't there, and I couldn't say goodbye 🙂
